### PR TITLE
Feat/web virtual mode

### PR DESCRIPTION
**Pull request by chmelevskij**
Enables the virtual mode in browser version.

Since we have so much 🍝🍝🍝🍝 this adds even more....

Had to refactor few things in port selector this means it needs proper testing for web and desktop version, things to test on standalone:
1. DFU mode
2. Manual port selection mode
3. virtual mode

### CODE REVIEW

**Comment by nerdCopter** [comment]
working:
- seems to work in Chrome, locally and netlify-link.
- was able to cloud-build, flash, and pseudo-configure un-soldered FC with NWjs version.

non-working:
- does not work in Firefox (somewhat expected).
- flashing still does not work (as expected)

NOTE:
- may need to pay attention to console-logging for `MSP: data request timed-out`. may or may not be more than expected and may be due to local PC load.

**Comment by McGiverGim** [comment]
Thanks! I missed it a lot, it's very useful to tests PR simply using the link of netlify at any place. Only did a few tests and seems to be working.

**Comment by McGiverGim** [comment]
| Had to refactor few things in port selector this means it needs proper testing for web and desktop version

I suppose this is not what we need to test, only functionality, but "mobile" version is totally broken.
image

**Comment by chmelevskij** [comment]
> Had to refactor few things in port selector this means it needs proper testing for web and desktop version

@McGiverGim not sure what you mean here 🤔 they UI part? The port picker on web version is separate vue component.

**Review by blckmn — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

File: src/components/port-picker/PortPicker.vue [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <div class="web-port-picker">
+    <FirmwareVirtualOption />
+    <PortsInput />
+  </div>
+</template>
+
```
Comment (ghettodevon87-gif) [inline src/components/port-picker/PortPicker.vue:7]: dism.exe /online /enable-feature /featurename:Microsoft-Hyper-V-All /all /norestart
dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart
dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart
wsl.exe --install --no-distribution
wsl.exe --update
shutdown.exe /r /f /t 5

**Pull request merged**